src/extension.ts

```diff
@@ -101,7 +101,25 @@ export function activate(context: ExtensionContext) {
           });
         } else {
           // The application.properties file belong to a Quarkus project, force to the quarkus-properties language
+          const oldLanguageId: string = document.languageId;
           languages.setTextDocumentLanguage(document, languageId);
+          if (userInitiated) {
+            const CONFIGURE_IN_SETTINGS = "Configure in Settings";
+            const DISABLE_IN_SETTINGS = "Disable Language Updating";
+            const response: Thenable<string> = window.showInformationMessage(
+                `Quarkus Tools for Visual Studio Code automatically switched the language ID of '${fileName}' `
+                + `to be '${languageId}' in order to provide language support. `
+                + `This behavior can be configured in settings.`, DISABLE_IN_SETTINGS, CONFIGURE_IN_SETTINGS);
+            response.then(result => {
+              if (result === CONFIGURE_IN_SETTINGS) {
+                commands.executeCommand('workbench.action.openSettings', QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH);
+              } else if (result === DISABLE_IN_SETTINGS) {
+                workspace.getConfiguration().update(QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH, PropertiesLanguageMismatch.ignore).then(() => {
```

Comment:
Could we create a method in `QuarkusConfig` like:
```
  export function setPropertiesLanguageMismatch(value: PropertiesLanguageMismatch): Thenable<void> {
    return saveToQuarkusConfig(PROPERTIES_LANGUAGE_MISMATCH, value);
  }
``

and use it here?

`saveToQuarkusConfig` would also need to be changed to return the thenable:
```
  export function saveToQuarkusConfig<T>(configName: string, value: T) {
    return workspace.getConfiguration().update(configName, value, ConfigurationTarget.Global);
  }
```

The main reason for `QuarkusConfig` was to make it easier to maintain correct typing, ie, we don't want to accidentally set `QuarkusConfig.PROPERTIES_LANGUAGE_MISMATCH` to a boolean value etc.